crates/storage/provider/src/trie/mod.rs

```diff
@@ -400,6 +494,48 @@ impl DBTrieLoader {
 
         Ok(hashed_changes)
     }
+
+    /// Returns a Merkle proof of the given account, plus its storage root hash.
+    pub fn generate_acount_proof<'tx, 'itx>(
+        &self,
+        tx: &'tx impl DbTx<'itx>,
+        root: H256,
```

Comment:
This is a weird API due to requiring the root to be passed by the user every time. Can we avoid that somehow?

Reply:
Maybe we can make the `DBTrieLoader` receive the current root on instantiation in a similar way to the database wrappers we use inside the module (i.e. with `new(root)` and `empty()` methods).

Reply:
Yeah I like that.

Reply:
Opened https://github.com/paradigmxyz/reth/issues/1743. Should be a small change, and a good first issue.